### PR TITLE
NO-ISSUE Add logs on error in controllers

### PR DESCRIPTION
**Pull request by rollandf**
Signed-off-by: Fred Rolland <frolland@redhat.com>

### CODE REVIEW

File: internal/controller/controllers/agent_controller.go [hunk 1]
```diff
@@ -131,10 +131,14 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 	if err = r.Get(ctx, kubeKey, clusterDeployment); err != nil {
 		if k8serrors.IsNotFound(err) {
 			// Delete the agent, using a finalizer with pre-delete to deregister the host.
+			log.Infof("Cluster Deployment name: %s namespace: %s not found, deleting Agent",
```
Comment (nmagnezi) [inline internal/controller/controllers/agent_controller.go:134]: hmm..
Not a comment about this log but, now that I look at it, maybe this should actually try to fetch ACI and not CD? as we do clear up agents per ACI Finalizer.

Comment (nmagnezi) [reply internal/controller/controllers/agent_controller.go:134]: @filanov what do you think? ☝️

File: internal/controller/controllers/agent_controller.go [hunk 2]
```diff
@@ -155,6 +161,7 @@ func (r *AgentReconciler) Reconcile(origCtx context.Context, req ctrl.Request) (
 	if host == nil {
 		// Host is not a part of the cluster, which may happen with newly created day2 clusters.
 		// Delete the agent, using a finalizer with pre-delete to deregister the host.
+		log.Infof("Host not found in Cluster ID:%s deleting Agent", string(*cluster.ID))
```
Comment (nmagnezi) [inline internal/controller/controllers/agent_controller.go:164]:
```suggestion
		log.Infof("Host not found in Cluster ID: %s deleting Agent", string(*cluster.ID))
```

File: internal/controller/controllers/agent_controller.go [hunk 3]
```diff
@@ -638,6 +647,7 @@ func (r *AgentReconciler) updateIfNeeded(ctx context.Context, log logrus.FieldLo
 			if errors.Is(err, gorm.ErrRecordNotFound) {
 				err = common.NewApiError(http.StatusNotFound, err)
 			}
+			log.WithError(err).Errorf("Failed to Aprove Agent")
```
Comment (nmagnezi) [inline internal/controller/controllers/agent_controller.go:650]:
```suggestion
			log.WithError(err).Errorf("Failed to approve Agent")
```

**Review by nmagnezi — CHANGES_REQUESTED** [review]

File: internal/controller/controllers/agent_controller.go (hunk 1 above)
Comment (filanov) [reply internal/controller/controllers/agent_controller.go:134]: maybe both, there maybe a case that only ACI is deleted right?

**Review by filanov — APPROVED** [review]

**Pull request merged**